cigar/cigar.go

```diff
@@ -3,22 +3,59 @@
 package cigar
 
 import (
-	"fmt"
-	"github.com/vertgenlab/gonomics/numbers/parse"
 	"log"
-	"unicode"
+	"strconv"
+	"strings"
+
+	"github.com/vertgenlab/gonomics/exception"
+	"github.com/vertgenlab/gonomics/numbers/parse"
 )
 
 // Cigar contains information on the runLength, operation, and DNA sequence associated with a particular cigar character.
 type Cigar struct {
 	RunLength int
-	Op        rune
+	Op        byte
+}
+
+// Cigar operation runes as defined in the SAM format specification to represent the different alignment sequences
+const (
+	Match     byte = 'M' // Alignment match (can be a sequence match or mismatch)
+	Insertion byte = 'I' // Insertion to the reference
+	Deletion  byte = 'D' // Deletion from the reference
+	Ns        byte = 'N' // Skipped region from the reference
+	SoftClip  byte = 'S' // Soft clipping (clipped sequences present in SEQ)
+	HardClip  byte = 'H' // Hard clipping (clipped sequences NOT present in SEQ)
+	Padded    byte = 'P' // Padding (silent deletion from padded reference)
+	Equal     byte = '=' // Sequence match
+	Mismatch  byte = 'X' // Sequence mismatch
+	Unmapped  byte = '*' // Represents an unmapped read in a SAM file.
+)
+
+// OpTable is a slice containing the runes representing CIGAR operation types
+var OpTable []byte = []byte{Match, Insertion, Deletion, Ns, SoftClip, HardClip, Padded, Equal, Mismatch}
+
+// Uint32Table is a map that provides a numeric representation of each CIGAR operation byte
+var Uint32Table = map[byte]uint32{
+	Match:     0,
+	Insertion: 1,
+	Deletion:  2,
+	Ns:        3,
+	SoftClip:  4,
+	HardClip:  5,
+	Padded:    6,
+	Equal:     7,
+	Mismatch:  8,
+}
+
+// IsUnmapped checks if provided cigars are unmapped. Note: Handle additional unmapped scenarios if needed.
+func IsUnmapped(cigars []Cigar) bool {
+	return len(cigars) == 0 // (empty slice)
```

Comment:
so if the cigar struct has a '*', it will return that the read is mapped? Should we switch the function to handle the length check and the `operation == *` check